lib/ansible/modules/crypto/openssl_privatekey.py

```diff
@@ -285,9 +287,13 @@ def generate(self, module):
                             self.mode = AnsibleModule._symbolic_mode_to_octal(st, self.mode)
                         except ValueError as e:
                             module.fail_json(msg="%s" % to_native(e), exception=traceback.format_exc())
-                os.chmod(self.path, self.mode)
                 privatekey_file = os.open(self.path, os.O_WRONLY | os.O_CREAT | os.O_TRUNC, self.mode)
-                os.write(privatekey_file, privatekey_data)
```

Comment:
This line should still be there (though probably one line down), while...

Reply:
Sorry for that I have reverted `os.write()`